src/Forms/SearchableDropdownTrait.php

```
@@ -0,0 +1,562 @@
+     * will be remvoved in a future version
+     */
+    public function setPlaceholder(string $placeholder): static
+    {
+        $this->placeholder = $placeholder;
+        // SearchableDropdownField will have the setHasEmptyDefault() method from SingleSelectField
+        if (method_exists($this, 'setHasEmptyDefault')) {
+            $this->setHasEmptyDefault(true);
+        }
+        return $this;
+    }
+
+    /**
+     * Get the search context to use
+     * If a search context has been set via setSearchContext() that will be used
+     * Will fallback to using the dataobjects default search context if a sourceList has been set
+     * Otherwise will return null
+     */
+    public function getSearchContext(): ?SearchContext
+    {
+        if ($this->searchContext) {
+            return $this->searchContext;
+        }
+        if ($this->sourceList) {
+            $dataClass = $this->sourceList->dataClass();
+            /** @var DataObject $obj */
+            $obj = $dataClass::create();
+            return $obj->getDefaultSearchContext();
+        }
+        return null;
+    }
+
+    /**
+     * Set the search context to use instead of the dataobjects default search context
+     *
+     * Calling this will also call setUseSearchContext(true)
+     */
+    public function setSearchContext(?SearchContext $searchContext): static
+    {
+        $this->searchContext = $searchContext;
+        $this->setUseSearchContext(true);
+        return $this;
+    }
+
+    /**
+     * Get whether to use a dynamic placeholder if a normal placeholder is not set
+     */
+    public function getUseDynamicPlaceholder(): bool
+    {
+        return $this->useDynamicPlaceholder;
+    }
+
+    /**
+     * Set whether to use a dynamic placeholder if a normal placeholder is not set
+     */
+    public function setUseDynamicPlaceholder(bool $useDynamicPlaceholder): static
+    {
+        $this->useDynamicPlaceholder = $useDynamicPlaceholder;
```

If setting this to true, should we clear the emptystring and placeholder values? Or should we instead throw an exception in `getPlaceholder()` if there is either an emptystring or placeholder value, but this is set to true?

---

I don't think either is necessary, there's docblocks on the methods roughly explaning how things work, I don't think developers will struggle with this so there's not real need to try and be extra helpful here